inhouse_bot/matchmaking_logic/evaluate_game.py

```diff
@@ -1,37 +1,66 @@
-import itertools
-import math
+import inhouse_bot.common_utils.lol_api.tasks as lol
+from typing import List
+from inhouse_bot.database_orm import Game, GameParticipant
+from inhouse_bot.dataclasses import GameInfo
+from sqlalchemy import BigInteger
+import logging
 
-import trueskill
 
-from inhouse_bot.database_orm import Game
+async def get_team_mmr(team: List[GameParticipant]) -> int:
+    mmrValues = {
+        "IRONIV": 200,
+        "IRONIII": 400,
+        "IRONII": 800,
+        "IRONI": 1000,
+        "BRONZEIV": 1100,
+        "BRONZEIII": 1150,
+        "BRONZEII": 1250,
+        "BRONZEI": 1300,
+        "SILVERIV": 1400,
+        "SILVERIII": 1450,
+        "SILVERII": 1550,
+        "SILVERI": 1600,
+        "GOLDIV": 1700,
+        "GOLDIII": 1750,
+        "GOLDII": 1850,
+        "GOLDI": 1900,
+        "PLATINUMIV": 2000,
+        "PLATINUMIII": 2050,
+        "PLATINUMII": 2150,
+        "PLATINUMI": 2200,
+        "DIAMONDIV": 2300,
+        "DIAMONDIII": 2350,
+        "DIAMONDII": 2450,
+        "DIAMONDI": 2500,
+        "MASTERI": 2750,
+        "GRANDMASTERI": 3000,
+        "CHALLENGERI": 3250,
+    }
 
+    mmr = 0
+    for gameparticipant in team:
+        summoner = await lol.get_summoner_by_puuid(
+            str(gameparticipant.player.summoner_puuid)
+        )
+        playerRankInfo = await lol.get_summoner_rank_info_by_id(summoner.id)
+        if playerRankInfo:
+            rankString = playerRankInfo["tier"] + playerRankInfo["rank"]
+            value = mmrValues[rankString]
+            mmr += value
+        else:
+            # We assume players without rank play at a silver level
+            # (TODO: Silver 4 can still be too high for unranked players, consider going to bronze/iron if matchmaking is really bad)
+            mmr += mmrValues["SILVERIV"]
```

Comment:
Does the the API returns data about past seasons potentially?

If not I'd make this 1500 (SILVERII) which is the default rating you get when you start playing ranked.

Reply:
The riot api unforunately does not give past season league data. 

Alternate solutions might be to rely on third-party websites' data (though many don't allow scraping), or having some mechanism for an admin to manually input a player's previous season rankings that then some formula could utilize in this computation.


> If not I'd make this 1500 (SILVERII) which is the default rating you get when you start playing ranked.

I assumed the default mmr here was more of a best guess. Do we know if Riot gives fresh players a silver 2 mmr?

Reply:
The values between silver 4 and silver 2 here are almost nothing, I also believe its better to shoot for a lower value, in the event someone is brand new to league and wants to partake in the inhouses, we can assume that they would be closer to bronze compared to gold for their performance(unless just gifted)